libs/net/dal/Services/ContentService.cs

```diff
@@ -150,6 +160,249 @@ public IPaged<Content> Find(ContentFilter filter, bool asNoTracking = true)
+                .Field(p => p.UpdatedOn)
+                .GreaterThanOrEquals(startOn)
+                .LessThanOrEquals(endOn)));
+        }
+        else if (filter.UpdatedStartOn.HasValue)
+        {
+            var startOn = filter.UpdatedStartOn.Value.ToUniversalTime().ToString("s") + "Z";
+            filterQueries.Add(s => s.DateRange(m => m
+                .Field(p => p.UpdatedOn)
+                .GreaterThanOrEquals(startOn)));
+        }
+        else if (filter.UpdatedEndOn.HasValue)
+        {
+            var endOn = filter.UpdatedEndOn.Value.ToUniversalTime().ToString("s") + "Z";
+            filterQueries.Add(s => s.DateRange(m => m
+                .Field(p => p.UpdatedOn)
+                .LessThanOrEquals(endOn)));
+        }
+
+        if (filter.PublishedOn.HasValue)
+        {
+            var publishedOn = filter.PublishedOn.Value.ToUniversalTime().ToString("s") + "Z";
+            filterQueries.Add(s => s.Match(m => m.Field(p => p.PublishedOn).Query(publishedOn)));
+        }
+
+        if (filter.PublishedStartOn.HasValue && filter.PublishedEndOn.HasValue)
+        {
+            var publishedStartOn = filter.PublishedStartOn.Value.ToUniversalTime().ToString("s") + "Z";
+            var publishedEndOn = filter.PublishedEndOn.Value.ToUniversalTime().ToString("s") + "Z";
+            filterQueries.Add(s => s.DateRange(m => m
+                .Field(p => p.PublishedOn)
+                .GreaterThanOrEquals(publishedStartOn)
+                .LessThanOrEquals(publishedEndOn)));
+        }
+        else if (filter.PublishedStartOn.HasValue)
+        {
+            var publishedStartOn = filter.PublishedStartOn.Value.ToUniversalTime().ToString("s") + "Z";
+            filterQueries.Add(s => s.DateRange(m => m
+                .Field(p => p.PublishedOn)
+                .GreaterThanOrEquals(publishedStartOn)));
+        }
+        else if (filter.PublishedEndOn.HasValue)
+        {
+            var publishedEndOn = filter.PublishedEndOn.Value.ToUniversalTime().ToString("s") + "Z";
+            filterQueries.Add(s => s.DateRange(m => m
+                .Field(p => p.PublishedOn)
+                .LessThanOrEquals(publishedEndOn)));
+        }
+
+        if (filter.UserId.HasValue)
+        {
+            filterQueries.Add(s => s.Term(t => t.OwnerId, filter.UserId.Value));
+        }
+
+        if (filter.Status.HasValue)
+        {
+            filterQueries.Add(s => s.Match(m => m.Field(p => p.Status).Query(filter.Status.Value.ToString())));
+        }
+
+        var response = await _client.SearchAsync<Content>(s => {
+            var result = s
+                .Pretty()
+                .Index(_unpublishedIndex)
+                .From((filter.Page - 1) * filter.Quantity)
+                .Size(filter.Quantity)
+                .Query(q => q.Bool(b => b
+                    .Filter(filterQueries)
+                    .Should(shouldQueries)))
+                ;
+
+            if (filter.Sort.Any())
```

Comment:
This will not work.  You can't sort after a search.  Your paging results will be horrible.

Reply:
The sort will be applied before searching, and it seems to be working fine. Please let me know if this is not the case, thanks.

Reply:
You are performing an asynchronous search prior to the sort.  Which means requesting any result will not return correct results.  

Think about what sorting does to results and how it is required for paging.

You can test this by creating content that when sorted shows up in the next page.

If you are saying the sort is correctly applied, it would be extremely odd implementation by Elasticsearch.  As it would imply the async search does not perform the search when you awaited it.

Reply:
It seems in this case sorting will be applied first before doing the search as we pass the func parameter which returns a SearchDescriptor<Content> to the ElasticClient's search function - SearchAsync before it being implemented, so in this case the sort should already been included before the search. Thanks.